src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -353,12 +389,14 @@ bool AdaptDispatch::CursorRestoreState()
     auto row = savedCursorState.Row;
     const auto col = savedCursorState.Column;
 
-    // If the origin mode is relative, and the scrolling region is set (the bottom is non-zero),
-    // we need to make sure the restored position is clamped within the margins.
-    if (savedCursorState.IsOriginModeRelative && _scrollMargins.Bottom != 0)
+    // If the origin mode is relative, we need to make sure the restored
+    // position is clamped within the margins.
+    if (savedCursorState.IsOriginModeRelative)
```

Comment:
I'm curious why we don't need to check any longer that there _are_ margins -- what overarching change made this safe to do both with and without margins?

Reply:
Now we have the `_GetVerticalMargins` helper function which always returns a valid set of margins - if margins haven't been set, it'll return the margins of the viewport itself. In this particular case it means we'd be clamping the row to the viewport, which is essentially a noop. But in general it makes things cleaner not having to care whether the margins are set or not.

Reply:
Ah, right -- you called this out in the PR body. Thanks!